src/function/memoize/__tests__/deep.js

```diff
@@ -0,0 +1,68 @@
+import memoizeDeep from '../deep';
+
+describe('src/memoize/deep', () => {
+    const prepare = () => {
+        const fn = jest.fn((...args) => args);
+        const memoizedFn = memoizeDeep(fn);
+
+        return {
+            fn,
+            memoizedFn
+        };
+    };
+
+    describe('with single primitive argument', () => {
+        it('returns cached value', () => {
+            const { fn, memoizedFn } = prepare();
+
+            expect(memoizedFn(1)).toBe(memoizedFn(1));
+            expect(fn).toHaveBeenCalledTimes(1);
+        });
+        it('return not cached value', () => {
+            const { fn, memoizedFn } = prepare();
+
+            expect(memoizedFn(1)).not.toBe(memoizedFn(2));
+            expect(fn).toHaveBeenCalledTimes(2);
+        });
+
+        it('returns not cached value for shallow equally string and number arguments', () => {
+            const { fn, memoizedFn } = prepare();
+
+            expect(memoizedFn(1)).not.toBe(memoizedFn('1'));
+            expect(fn).toHaveBeenCalledTimes(2);
+        });
+    });
+    describe('with multi primitive argument', () => {
+        it('returns cached value', () => {
+            const { memoizedFn, fn } = prepare();
+
+            expect(memoizedFn(1, 2)).toBe(memoizedFn(1, 2));
+            expect(fn).toHaveBeenCalledTimes(1);
+        });
+        it('returns not cached value', () => {
+            const { memoizedFn, fn } = prepare();
+
+            expect(memoizedFn(1, 2)).not.toBe(memoizedFn(1, 2, 3));
+            expect(fn).toHaveBeenCalledTimes(2);
+        });
+    });
+
+    describe('with multple non-primitive arguments', () => {
```

Comment:
test description says its for multiple arguments but actually testing only for single argument

Reply:
fixed